driver/src/main/java/org/neo4j/driver/internal/handlers/PullAllResponseHandler.java

```diff
@@ -66,10 +67,11 @@
     private CompletableFuture<Record> recordFuture;
     private CompletableFuture<Throwable> failureFuture;
 
-    public PullAllResponseHandler( Statement statement, RunResponseHandler runResponseHandler, Connection connection )
+    public PullAllResponseHandler( Statement statement, RunResponseHandler runResponseHandler, String resultConsumedAfterMetadataKey, Connection connection )
```

Comment:
👍 I'll give this a try. Might especially be valuable if we add another response metadata flag

Reply:
Made the metadata extraction utility versioned instead of passing string keys around. Creating classes felt a little to much - 3 new classes would need to be created just to override 2 string keys